modules/autotagging-commons/common/src/main/java/org/opensearch/rule/service/IndexStoredRulePersistenceService.java

```diff
@@ -129,6 +148,90 @@ void handleGetRuleResponse(List<SearchHit> hits, ActionListener<GetRuleResponse>
         listener.onResponse(new GetRuleResponse(ruleMap, nextSearchAfter));
     }
 
+    /**
+     * Entry point for the update rule api logic in persistence service.
+     * @param request - The UpdateRuleRequest
+     * @param listener - ActionListener for UpdateRuleResponse
+     */
+    public void updateRule(UpdateRuleRequest request, ActionListener<UpdateRuleResponse> listener) {
+        String ruleId = request.get_id();
+        FeatureType featureType = request.getFeatureType();
+        try (ThreadContext.StoredContext context = getContext()) {
+            QueryBuilder query = queryBuilder.from(new GetRuleRequest(ruleId, new HashMap<>(), null, featureType));
+            getRuleFromIndex(ruleId, query, null, new ActionListener<>() {
+                @Override
+                public void onResponse(GetRuleResponse getRuleResponse) {
+                    if (getRuleResponse == null || getRuleResponse.getRules().isEmpty()) {
+                        listener.onFailure(new ResourceNotFoundException("Rule with ID " + ruleId + " not found."));
+                        return;
+                    }
+                    Rule updatedRule = RuleUtils.composeUpdatedRule(getRuleResponse.getRules().get(ruleId), request, featureType);
```

Comment:
Having static methods make the unit testing harder since we can't mock them in OpenSearch